atomic_reactor/plugins/check_and_set_platforms.py

```diff
@@ -115,11 +113,14 @@ def has_enabled_hosts(platform: str) -> bool:
                   ' {}'.format(defined_but_disabled)
             raise RuntimeError(msg)
 
-        final_platforms = self._limit_platforms(enabled_platforms)
-        self.log.info("platforms in limits : %s", final_platforms)
-        if not final_platforms:
-            self.log.error("platforms in limits are empty")
-            raise RuntimeError("No platforms to build for")
+        if user_platforms_override:
+            final_platforms = platforms
```

Comment:
This allows users  to enable a platform that is not defined in the config, do we want that?

It worked like that before, but since we're already changing the logic, maybe we should fix it?

Reply:
I don't think it will be skipped, I think it will fail like this
https://github.com/containerbuildsystem/atomic-reactor/blob/20e8170fec6d3a0d21c0eec674d9a553b26ba4b9/atomic_reactor/utils/remote_host.py#L669-L671

https://github.com/containerbuildsystem/atomic-reactor/blob/20e8170fec6d3a0d21c0eec674d9a553b26ba4b9/atomic_reactor/tasks/binary_container_build.py#L130-L134

Reply:
https://github.com/containerbuildsystem/atomic-reactor/blob/20e8170fec6d3a0d21c0eec674d9a553b26ba4b9/atomic_reactor/tasks/binary_container_build.py#L70

Reply:
also that "feature" is used by fedora

Reply:
> https://github.com/containerbuildsystem/atomic-reactor/blob/20e8170fec6d3a0d21c0eec674d9a553b26ba4b9/atomic_reactor/tasks/binary_container_build.py#L70
> 
> if platform isn't enabled / defined it will just return

On that line, `enabled_platforms` is the return value of the check_and_set_platforms plugin. If this plugin returns a platform which is not defined in the config, the build will fail with "Failed to acquire a build slot ..."